tests/test_algos/test_algos.py

```diff
@@ -30,70 +30,106 @@ def mock_env_and_destroy(devices):
         dist.destroy_process_group()
 
 
-def check_checkpoint(algo: str, target_keys: set):
+def check_checkpoint(ckpt_path: str, target_keys: set, checkpoint_buffer: bool = True):
     fabric = Fabric(accelerator="cpu")
     if not _is_using_cli():
         fabric.launch()
-    project_root = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
-    ckpt_path = f"{project_root}/logs/{algo}/"
-    experiment_list = sorted(os.listdir(ckpt_path))
-    assert len(experiment_list) > 0
-    ckpt_path += experiment_list[-1] + "/"
-    ckpt_path += os.listdir(ckpt_path)[-1] + "/version_0/checkpoint/"
-    assert len(os.listdir(ckpt_path)) == 1
-    state = fabric.load(ckpt_path + os.listdir(ckpt_path)[-1])
+
+    # check the presence of the checkpoint
+    assert os.path.isdir(ckpt_path)
+    state = fabric.load(ckpt_path + "/" + os.listdir(ckpt_path)[-1])
+
+    # the keys in the checkpoint must match with the expected keys
     ckpt_keys = set(state.keys())
-    assert len(ckpt_keys.intersection(target_keys)) == len(ckpt_keys)
+    assert len(ckpt_keys.intersection(target_keys)) == len(ckpt_keys) == len(target_keys)
+
+    # if checkpoint_buffer is false, then "rb" cannot be in the checkpoint keys
+    assert checkpoint_buffer or "rb" not in ckpt_keys
```

Comment:
Maybe we should remove the test folder after the check. What do you think?

Reply:
Looks fine for me, I will also change the assumption that the root of the log directory is always `logs/<algo_name>`